zrml/prediction-markets/src/tests.rs

```diff
@@ -138,15 +143,17 @@ fn it_allows_to_deploy_a_pool() {
             100 * BASE,
         ));
 
-        assert_ok!(Shares::wrap_native_currency(
+        assert_ok!(Balances::transfer(
             Origin::signed(BOB),
+            <Test as crate::Trait>::ModuleId::get().into_account(),
             100 * BASE
         ));
+        assert_ok!(Shares::generate(Asset::Ztg, &BOB, 100 * BASE));
```

Comment:
This is roughly the same as `wrap_native_currency`, but sure if the semantics are correct though

Reply:
Is `Asset::Ztg` distinct from Balances? Like does `orml_tokens` still have a separate storage for it from balances?

Reply:
If so, can we use `orml_currencies` to replace `zrml_shares` and implement a trait for it to have `destroy_all`?

Reply:
Implementation-wise, `PredictionMarkets::Shares` has a separated storage attached to `zrml_shares::Module` just like the old one had, the only difference is the usage of `orml_tokens` instead of `pallet_balances`.
`zrml_shares` is now just a wrapper around the `orml_traits` trait methods, therefore, it doesn't seem necessary any more.

Reply:
Yep makes sense to remove `zrml_shares` now

Reply:
But, can we make Balances ZTG interoperable with tokens in pallets such as Swaps without needing to first convert Balances ZTG storage to Tokens ZTG storage?

Reply:
Both have separated storages, which makes sense since they are two different things. The only way to seamlessly exchange tokens between `Balances` and `Tokens` is, in my opinion, to have the same balance type `Asset`.
With this PR, every token `Ztg` token took from the balance must sync with the pool accordingly by issuing command with `Asset::Ztg`.

Reply:
That all makes sense, but my sticking point is the following:

- ZTG is Balances
- Swaps works with Tokens
- Using ZTG in Swaps means first creating the separate instance of "ZTG token" on Tokens

Please correct me if my assumption is wrong.

We can make it so that Swaps works with "ZTG Balances" without requiring to turn into "ZTG token" by using an implementation that calls either Balances or Tokens based on the provided `Asset` type. This is what ORML `MultiCurrency` trait and `orml_currencies` does. So I think we can finish by 1) removing Shares, 2) use `MultiReservableCurrency` everywhere `ReservableShares` was used 3) adding `orml_currencies`. WDYT?

Reply:
Things are clear now, thanks for the explanation. I will use `orml_currencies` for pools but it might also be extended to prediction markets and order books

Just for the record, `orml_currencies` methods are roughly composed by an addition storage and
```rust
if currency_id == T::GetNativeCurrencyId::get() {
	... stuff
} else {
	... more stuff
}
```
to track what is and isn't `native`. Not sure if the additional capacity and processing overhead will be beneficial. What will change? Faster native queries?

Reply:
It's O(1),  just adding a single storage read to the methods. It's definitely beneficial, both to the end user who doesn't need to manage two different types of ZTG, and for the chain's storage size since we only have 1 storage for ZTG instead of 2 (which could be at worst double the storage size, for no reason). 

Not sure if that answers your question or not, so let me know if it didn't.

Reply:
Thank you very much for the answer, I understand now.